games/xworld/xworld_simulator.cpp

```diff
@@ -154,15 +154,21 @@ int XWorldSimulator::game_over() {
 float XWorldSimulator::take_action(const StatePacket& actions) {
     TeachingEnvironment::take_action(actions);
     last_action_ = "";
-    std::string agent_sent = *(actions.get_buffer("pred_sentence")->get_str());
-    // TODO: Currently assume only one agent for speak action
-    record_agent_sent_in_buffer(agent_sent);
-    last_action_ += "speak(" + agent_sent + ")";
-    // update message box
-    history_messages_.push_back("[Reply] Learner: " + agent_sent);  // add token
-    update_message_box_on_screen();
+
+    if (task_mode_ != "arxiv_lang_acquisition") {
```

Comment:
maybe task_mode_ == "arxiv_interactive" || task_mode_ == "one_channel" is more straightforward?

Reply:
Done.